File: Factorials/factorials.py
def factorial(a: int):
    """
    handle edge case
    a == 0 or 1 or -ve number
    """
    if a == 0 or a == 1:
        return 1

    if a < 0:
        return "Negative Integer is not allowed"

    # recursive call
    return a * factorial(a - 1)


if __name__ == "__main__":
    n = int(input("Enter the positive integer value: "))
    answer = factorial(n)
    print(f"{n}! = {answer}")

File: Factorials/factorials.c
#include<stdio.h>  
  
unsigned long long factorial(int n)  
{  
  if (n == 0)  
    return 1;  
  else  
    return(n * factorial(n-1));  
}  
    
void main()  
{  
  int number;  
  unsigned long long fact;  
  printf("Enter a number: ");  
  scanf("%d", &number);   
    
  fact = factorial(number);  
  printf("Factorial of %d is %llu\n", number, fact);
}  
